src/main/java/com/cgdp/library/dto/book/BookDTO.java

```diff
@@ -0,0 +1,24 @@
+package com.cgdp.library.dto.book;
+
+import com.cgdp.library.entity.AuthorEntity;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import java.time.LocalDate;
+
+@Data
+public class BookDTO {
+
+    @NotNull
+    private Long id;
+    @NotBlank
+    private String title;
+    @NotNull
+    private Long authorId;
+    @NotNull
+    private LocalDate publicationYear;
+    @NotBlank
```

Comment:
This field should be optional, does this annotation allow that? Also create custom annotation to validate if is valid isbn

`public @interface ValidIsbn` 
https://www.baeldung.com/spring-mvc-custom-validator